js/framework/AnalyticsConstants.js

```diff
@@ -885,6 +959,190 @@ if (!OO.Analytics.EVENT_DATA)
     var checkAdErrorData = OO._.bind(checkDataType, this, "AdErrorData");
     this.error = checkAdErrorData(error, "error", ["string", "object"]);
   };
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#VideoPlayerCreatedData
+   * @classdesc Contains information about the player created event
+   * @property {string} playerCoreVersion The player core version
+   * @property {object} params The configuration metadata associated with the player
+   * (i.e. pcode, playerBrandingId, skin configuration, player configuration parameters)
+   * @property {string} embedCode The embed code of the asset attempting to play
+   * @property {string} playerUrl The url of the page containing the player
+   */
+  EVENT_DATA.VideoPlayerCreatedData = function(playerCoreVersion, params, embedCode, playerUrl)
+  {
+    var checkVideoPlayerCreatedData = OO._.bind(checkDataType, this, "VideoPlayerCreatedData");
+    this.playerCoreVersion = checkVideoPlayerCreatedData(playerCoreVersion, "playerCoreVersion", ["string"]);
+    this.params = checkVideoPlayerCreatedData(params, "params", ["object"]);
+    this.embedCode = checkVideoPlayerCreatedData(embedCode, "embedCode", ["string"]);
+    this.playerUrl = checkVideoPlayerCreatedData(playerUrl, "playerUrl", ["string"]);
+    this.pcode = checkVideoPlayerCreatedData(this.params.pcode, "pcode", ["string"]);
+  };
+
+  /**
+   * @public
+   * @class Analytics.EVENT_DATA#InitialPlayStartingData
+   * @classdesc Contains the information about the initial play starting event.
+   * @property {string} playerCoreVersion The player core version
+   * @property {number} timeSinceInitialPlay The time since the initial play request was made
+   * @property {boolean} autoplayed Boolean for if the video was autoplayed or not
+   * @property {boolean} hadPreroll Boolean for if the video had an ad play before it started
+   * @property {number} position The initial position of the playhead upon playback start. This includes 
+   *   midrolls that play before content due to an initial playhead time > 0
+   * @property {string} plugin The video plugin used for playback
+   * @property {string} technology The browser technology used - HTML5, Flash, Mixed, or Other
+   * @property {string} encoding The stream encoding type, i.e. MP4, HLS, Dash, etc.
```

Comment:
Should this be a string or object? Code below checks if it is an object:

this.encoding = checkInitialPlayStartingData(encoding, "encoding", ["object"]);

Reply:
Should be string, fixed